grafana-dashboard.tf

```diff
@@ -0,0 +1,36 @@
+resource "random_integer" "graph_id" {
+  min     = 1000
+  max     = 999999
+  count = length(local.apiary_data_buckets)
+}
+
+data "template_file" "graphs" {
+  template = file("${path.module}/templates/graph.tpl")
+  count = length(local.apiary_data_buckets)
+  vars = {
+    bucket_name = local.apiary_data_buckets[count.index]
+    title_bucket_name = local.apiary_managed_schema_names_replaced[count.index]
+    graph_id = random_integer.graph_id.*.result[count.index]
+  }
+}
+
+data "template_file" "dashboard" {
+  template = file("${path.module}/templates/dashboards.tpl")
+  vars = {
+    panels = "[${join(",", data.template_file.graphs.*.rendered)}]"
+  }
+}
+
+resource "kubernetes_config_map" "grafana-dashboard" {
```

Comment:
What if someone is using this open-source module in a deployment where they don't have Grafana?  Should we be generating all this if there is no Grafana?

Reply:
How would we check if Grafana is running? 

I asked Raj a similar question (If users of apiary-data-lake haven't configured Datawatch as a source for Grafana, none of this will work), his opinion was that apiary-data-lake assumes a few things are in place, such as kubernetes provider, grafana is set up and is configured to pick up the config maps.

Reply:
I would say if Apiary is deployed to EKS (not ECS), _then_ we assume Grafana is in place.  So it seems like all of this can be constrained by `count = "${var.hms_instance_type == "k8s" ? 1 : 0}"`

Reply:
Assumption is that when we are deploying k8s, evironment is setup accordingly. we can add this to k8s requirements.

Reply:
@rpoluri  where are the requirements? Where should I add the grafana requirements?